tests/test_encoding.py

```diff
@@ -26,22 +26,21 @@ def test_encoding2(v):
     assert not v.found_dead_code_or_error
 
 
-def test_non_utf8_encoding(v, tmpdir):
+def test_non_utf8_encoding(v, tmp_path):
     code = ""
     name = "non_utf8"
-    non_utf_8_file = str(tmpdir.mkdir(name).join(name + ".py"))
+    non_utf_8_file = tmp_path / (name + ".py")
     with open(non_utf_8_file, mode="wb") as f:
         f.write(codecs.BOM_UTF16_LE)
         f.write(code.encode("utf_16_le"))
-    v.scavenge([f.name])
+    v.scavenge([str(non_utf_8_file)])
```

Comment:
Let's pass strings to scavenge() and scan() in the tests.

Reply:
This comment is confusing. What I mean is: in the tests, let's pass a Path object if we already have one (like in the code above) and let's pass strings in the other cases below (where we currently create Path objects just to comply with the API).

Reply:
Good point. Will do.